src/testModel.ts

```diff
@@ -89,26 +90,38 @@ export class TestModel extends DisposableBase {
     this._collection = collection;
     this.config = { ...playwrightInfo, workspaceFolder, configFile };
     this._useLegacyCLIDriver = playwrightInfo.version < 1.44;
-    this._playwrightTest =  this._useLegacyCLIDriver ? new PlaywrightTestCLI(this._vscode, this, collection.embedder) : new PlaywrightTestServer(this._vscode, this, collection.embedder);
+    if (this._useLegacyCLIDriver) {
+      this._playwrightTest = new PlaywrightTestCLI(this._vscode, this, collection.embedder);
+      this._traceViewerFactory = new TraceViewerFactory(collection, this.config);
+    } else {
+      this._playwrightTest = new PlaywrightTestServer(this._vscode, this, collection.embedder);
+      this._traceViewerFactory = new TraceViewerFactory(collection, this.config, this._playwrightTest);
+    }
     this.tag = new this._vscode.TestTag(this.config.configFile);
-    this._spawnTraceViewer = new SpawnTraceViewer(this._vscode, this._embedder.envProvider, this.config);
 
     this._disposables = [
       this._embedder.settingsModel.showTrace.onChange(() => this._closeTraceViewerIfNeeded()),
+      this._embedder.settingsModel.embeddedTraceViewer.onChange(() => this._closeTraceViewerIfNeeded()),
       this._collection.onUpdated(() => this._closeTraceViewerIfNeeded()),
     ];
   }
 
   traceViewer() {
-    if (!this._embedder.settingsModel.showTrace.get())
-      return;
-    if (this._spawnTraceViewer.checkVersion())
-      return this._spawnTraceViewer;
+    return this._traceViewer;
   }
 
-  _closeTraceViewerIfNeeded() {
-    if (this._collection.selectedModel() !== this || !this._embedder.settingsModel.showTrace.get())
-      this._spawnTraceViewer.close();
+  ensureTraceViewer(): TraceViewer | null {
+    if (this._traceViewer === undefined)
+      this._traceViewer = this._traceViewerFactory.create();
+
+    return this._traceViewer;
+  }
+
+  private _closeTraceViewerIfNeeded() {
+    if (this._traceViewer && this._collection.selectedModel() === this && this._traceViewerFactory.isSupported(this._traceViewer))
```

Comment:
This will never close spawned trace viewer since it is always supported.

Reply:
I was referring to changing the settings value. If you change the setting value nothing will happen.